components/custom_status/custom_status_emoji.tsx

```diff
@@ -1,27 +1,40 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
-import React from 'react';
+import React, {useEffect} from 'react';
 import {Tooltip} from 'react-bootstrap';
-import {useSelector} from 'react-redux';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {getCustomEmojis} from 'mattermost-redux/actions/emojis';
 
 import OverlayTrigger from 'components/overlay_trigger';
 import RenderEmoji from 'components/emoji/render_emoji';
 import {makeGetCustomStatus, isCustomStatusEnabled} from 'selectors/views/custom_status';
+import {isCustomEmojiEnabled} from 'selectors/emojis';
 import {GlobalState} from 'types/store';
 import Constants from 'utils/constants';
 
 interface ComponentProps {
     emojiSize?: number;
     showTooltip?: boolean;
     tooltipDirection?: 'top' | 'right' | 'bottom' | 'left';
+    spanStyle?: React.CSSProperties;
     emojiStyle?: React.CSSProperties;
     userID?: string;
     onClick?: () => void;
 }
 
 const CustomStatusEmoji = (props: ComponentProps) => {
+    const dispatch = useDispatch();
     const getCustomStatus = makeGetCustomStatus();
-    const {emojiSize, emojiStyle, showTooltip, tooltipDirection, userID, onClick} = props;
+    const {emojiSize, emojiStyle, spanStyle, showTooltip, tooltipDirection, userID, onClick} = props;
+
+    const customEmojiEnabled = useSelector(isCustomEmojiEnabled);
+    useEffect(() => {
+        if (customEmojiEnabled) {
+            dispatch(getCustomEmojis());
```

Comment:
This only gets the first page of custom emojis which doesn't work when there's too many emojis on the server. It also isn't debounced in any way, so we'll be calling it repeatedly for each CustomStatusEmoji mounted at once. There is a getCustomEmojiByName singular action, but I'd still be worried about potentially calling it multiple times if multiple components are mounted at once.

We don't have a great way to handle that sort of loading right now, but I think the best way we can do it is to load the custom emojis when the user profiles are loaded similar to how we call `loadStatusesForProfileLists`(https://github.com/mattermost/mattermost-webapp/blob/master/actions/user_actions.jsx#L44) in a bunch of places. We'll also want to only load the emoji if necessary too.

Reply:
@hmhealey What if we remove the server version check from `loadMe` in `mattermost-redux`?

Ref - https://github.com/mattermost/mattermost-redux/blob/041dc98731d1317c7d4f8014d79f81d1e9e80663/src/actions/users.ts#L227

Reply:
That isn't possible. Loading all custom emojis adds a lot of lag for people that have thousands of emojis which is why we switched to loading them on-demand. We went from doing it on page load to loading any unrecognized emojis when we received a batch of posts with them in it. In this case, our best bet is probably to do something similar but doing it when we receive a batch of users with custom statuses enabled.

Reply:
Since that fix will likely be more involved, we might want to consider moving it out of 5.33 and just merging the other fixes for now. I'm not sure what the full effects of not having the custom emoji for a status might be though.

Reply:
I agree. @chetanyakan as suggested here let's pull this specific bug fix from this PR so we can unblock the rest of these fixes. We are aiming to merge this before we cut the RC for v5.33 on wednesday

Reply:
Done @hmhealey @esethna